frameworks/hdfs/tools/docker-client/Dockerfile

```diff
@@ -1,37 +1,39 @@
 FROM ubuntu:16.04
 
-RUN apt-get -y update && \
-    apt-get -y upgrade && \
-    apt-get -y install wget software-properties-common
-
-RUN wget https://downloads.mesosphere.com/hdfs/assets/hadoop-2.6.0-cdh5.9.1-dcos.tar.gz && \
-    tar xf hadoop-2.6.0-cdh5.9.1-dcos.tar.gz && \
-    rm hadoop-2.6.0-cdh5.9.1-dcos.tar.gz
-
-RUN apt-get update && \
-    apt-get install -y default-jre && \
-    apt-get install -y \
-    curl \
-    dnsutils \
-    less \
-    vim
-
 ENV HADOOP_VERSION hadoop-2.6.0-cdh5.9.1
 ENV HDFS_SERVICE_NAME hdfs
 ENV JAVA_HOME /usr/lib/jvm/default-java
 
+RUN apt-get -y update && \
+    apt-get -y upgrade && \
+    apt-get -y install \
+      curl \
+      default-jre \
+      dnsutils \
+      less \
+      software-properties-common \
+      vim \
+      wget && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN wget https://downloads.mesosphere.com/hdfs/assets/$HADOOP_VERSION-dcos.tar.gz && \
+    tar xf $HADOOP_VERSION-dcos.tar.gz && \
+    rm $HADOOP_VERSION-dcos.tar.gz
+
+
 # the env var is required for a non-interactive install of krb5-user
-RUN export DEBIAN_FRONTEND=noninteractive && apt-get install -y krb5-user
-COPY krb5.conf /etc/krb5.conf
-RUN chown -R nobody /etc
+RUN export DEBIAN_FRONTEND=noninteractive && \
+    apt-get update && \
+    apt-get install -y krb5-user && \
+    apt-get clean
 
-COPY configure-environment.sh /hadoop-2.6.0-cdh5.9.1/configure-environment.sh
-COPY configure-hdfs.sh /hadoop-2.6.0-cdh5.9.1/configure-hdfs.sh
+COPY krb5.conf /etc/krb5.conf
 
-# working directory will be the HDFS distribution
-RUN chown -R nobody /hadoop-2.6.0-cdh5.9.1
+COPY configure-environment.sh /$HADOOP_VERSION/configure-environment.sh
+COPY configure-hdfs.sh /$HADOOP_VERSION/configure-hdfs.sh
 
-WORKDIR /hadoop-2.6.0-cdh5.9.1
+WORKDIR /$HADOOP_VERSION
 
 # this script will setup the HDFS configuration regardless of HDFS being kerberized or not
 CMD "./configure-environment.sh"
```

Comment:
If we specify the full path here, the image would also work when the user specifies a `-w` flag to the docker command?

Reply:
Makes sense, prefixed `CMD` with hadoop directory